operator/v1/types_ingress.go

```diff
@@ -536,6 +536,46 @@ type HostNetworkStrategy struct {
 	// +kubebuilder:validation:Optional
 	// +optional
 	Protocol IngressControllerProtocol `json:"protocol,omitempty"`
+
+	// httpPort is the port on the host which should be used to listen for
+	// HTTP requests. This field should be set when port 80 is already in use.
+	// The value should not coincide with the NodePort range of the cluster.
+	// When not specified the value defaults to 80.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=80
+	HTTPPort int32 `json:"httpPort"`
```

Comment:
CRD changes can be observed on a delay.  What does the operator do when this field is zero-value, not 80, because the kube-apiserver has not observed the default.

Reply:
We will have the same defaults also in the operator so if a 0 is observed it is defaulted there as well.

https://github.com/openshift/cluster-ingress-operator/pull/694/files#L392-L404